packages/core/src/parsers/linkProcessor.js

```diff
@@ -23,41 +41,114 @@ const defaultTagLinkMap = {
   const fullResourcePath = path.join(cwd, resourcePath);
-  const resourcePathFromRoot = utils.ensurePosix(path.relative(rootPath, fullResourcePath));
+  const resourcePathFromRoot = getResourcePathFromRoot(rootPath, fullResourcePath);
 
+  const linkAttribName = defaultTagLinkMap[node.name];
   node.attribs[linkAttribName] = path.posix.join(baseUrl || '/', resourcePathFromRoot);
 }
-
-function convertMdExtToHtmlExt(node) {
+function convertMdAndMbdExtToHtmlExt(node) {
   if (node.name === 'a' && node.attribs && node.attribs.href) {
+    const hasNoConvert = lodashHas(node.attribs, 'no-convert');
+    if (hasNoConvert) {
+      return;
+    }
     const { href } = node.attribs;
     const hasMdExtension = href.slice(-3) === '.md';
-    const hasNoConvert = lodashHas(node.attribs, 'no-convert');
-    if (hasMdExtension && !hasNoConvert) {
+    if (hasMdExtension) {
       const newHref = `${href.substring(0, href.length - 3)}.html`;
       node.attribs.href = newHref;
+      return;
+    }
+    const hasMbdExtension = href.slice(-4) === '.mbd';
+    if (hasMbdExtension) {
+      const newHref = `${href.substring(0, href.length - 4)}.html`;
+      node.attribs.href = newHref;
+    }
+  }
+}
+
+function validateIntraLink(node, cwf, config) {
+  // checks if the file indicated in the intra link exists
+  const originalResourcePath = getResourcePath(node, cwf, config.rootPath);
+  if (!originalResourcePath || utils.isUrl(originalResourcePath) || originalResourcePath.startsWith('#')) {
+    return;
+  }
+
+  let resourcePath = originalResourcePath;
+  // remove hash portion (if any) in the resourcePath
+  const resourcePathUrl = url.parse(resourcePath);
+  if (resourcePathUrl.hash) {
+    resourcePath = resourcePathUrl.path;
+  }
+
+  const tempResourcePath = resourcePath.substring(1); // remove the '/' first char
+
+  // check if the file is a Page source
+  // since this function is called after convertMdAndMbdExtToHtml,
+  // we can assume that page source mostly has .html extension (unless user indicates no-convert)
+  const hasHtmlExt = resourcePath.slice(-5) === '.html';
+  if (hasHtmlExt) {
+    const resourcePathWithNoExt = fsUtil.removeExtension(tempResourcePath);
+    const isPageSrc = config.addressablePagesSource.includes(resourcePathWithNoExt);
+    // if not pageSrc, it is a raw asset
+    if (isPageSrc) {
+      // Page source file exists
+      return;
+    }
+  }
+
+  // file is an asset
+
+  // checks if the asset file is an ignored file
+  const fileIgnore = ignore().add(config.ignore);
+  if (tempResourcePath && fileIgnore.ignores(tempResourcePath)) {
+    return;
+  }
+
+  // asset file is not ignored, check for the existence of asset using fs
+
+  // only dealing with absolute paths because relative paths were already converted to absolute
+  let fullResourcePath = path.join(config.rootPath, resourcePath);
+  if (config.baseUrl) {
+    fullResourcePath = fullResourcePath.replace(config.baseUrl, '');
```

Comment:
hmm shouldn't this be first and foremost? (replacing `tempResourcePath = ...`)
since the paths in `addressablePagesSource` are not prepended by the baseUrl

let's do it more safely as well, e.g. ``path.posix.relative(`${config.baseUrl}/`, resourcePath)``, in case the baseUrl appears in the middle of the path

Reply:
Hmm, that's true. My bad for missing this out. Will do as what you suggested!